app/controllers/das_controller.rb

```diff
@@ -1,123 +1,122 @@
-    end
+          @snps.each do |snp|
+            # there is only one user_snp for each snps
+            if @types != []
+              @single_user_snp = snp.user_snps.find_by_genotype_id(@genotype.id)
+              if @types.include? @single_user_snp.local_genotype
+                @tmp_user_snps << @single_user_snp
+              end
+            else
+              @tmp_user_snps << snp.user_snps.find_by_genotype_id(@genotype.id)
+            end
+          end
 
-    def sources
-        @users = User.all
-        response.headers["X-DAS-Version"] = "DAS/1.53E"
-        response.headers["X-DAS-Status"] = "200"
-        response.headers["X-DAS-Capabilities"] = "features/1.1; sources/1.0"
+          @user_snps << @tmp_user_snps
+          @has_start << false
+         end
+      end
+      # When everything went correctly, send back 200
+      response.headers['X-DAS-Status'] = '200'
+    else
+      # no chromosome or start/end, so get nothing
+      @user_snps = [] # @user.user_snps
+      @id = ''
+      @start = ''
+      @end  = ''
 
-        response.headers["X-DAS-Server"] = request.env["SERVER_SOFTWARE"].split(" ")[0]
- 
-        render :template => 'das/sources.xml.erb', :layout => false
+      # Bad command arguments (arguments invalid)
+      response.headers['X-DAS-Status'] = '402'
     end
-    
-    def startpoint
-      @user = User.find_by_id(params[:id])
-      response.headers["X-DAS-Version"] = "DAS/1.53E"
-      response.headers["X-DAS-Status"] = "200"
-      response.headers["X-DAS-Capabilities"] = "features/1.1; sources/1.0"
-      response.headers["X-DAS-Server"] = request.env["SERVER_SOFTWARE"].split(" ")[0]
-      
-      render :template => 'das/startpoint.xhtml.erb', :layout => false
-    end
-end
+    response.headers['X-DAS-Version'] = 'DAS/1.53E'
+    # Change these capabilities once we implement more
+    response.headers['X-DAS-Capabilities'] = 'features/1.1; sources/1.0'
+    # Put in the servername and version
+    response.headers['X-DAS-Server'] = request.env['SERVER_SOFTWARE'].split(' ')[0]
+    response.headers['Access-Control-Allow-Origin'] = '*'
+
+    render template: 'das/show.xml.erb', layout: false
+  end
 
+  def sources
+    @users = User.all
+    response.headers['X-DAS-Version'] = 'DAS/1.53E'
+    response.headers['X-DAS-Status'] = '200'
+    response.headers['X-DAS-Capabilities'] = 'features/1.1; sources/1.0'
 
+    response.headers['X-DAS-Server'] = request.env['SERVER_SOFTWARE'].split(' ')[0]
+    render template: 'das/sources.xml.erb', layout: false
+  end
+
+  def startpoint
+    @user = User.find_by_id(params[:id])
+    response.headers['X-DAS-Version'] = 'DAS/1.53E'
+    response.headers['X-DAS-Status'] = '200'
+    response.headers['X-DAS-Capabilities'] = 'features/1.1; sources/1.0'
+    response.headers['X-DAS-Server'] = request.env['SERVER_SOFTWARE'].split(' ')[0]
+
+    render template: 'das/startpoint.xhtml.erb', layout: false
+  end
+end
```

Comment:
I didn't want to go down this rabbit hole. I just fixed the indentation and made it work with the new schema.